### PR TITLE
OOP school library: add basic UI

### PR DESCRIPTION
**Pull request by joel-kalema**
- [ ] Your console app should behave in the same way is in the example.
- [ ] Create a app.rb file that will serve as your console app entry-point.
- [ ] In your main.rb define the entry point, this will be a method called main that is invoked at the end of your file.

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+    when 'y'
+      student = Student.new(age: age, name: name, parent_permission: true)
+    when 'n'
+      student = Student.new(age: age, name: name, parent_permission: false)
+    else
+      puts 'Invalid. Try again'
+    end
+    @people << student
+    puts 'Student created successfully!!'
+  end
```
Comment (julie-ify) [inline app.rb:70]: - [ ] The app crashes with the attached error message when I tried creating a student. Kindly fix this to make your app work as expected.
chrome_kijUpGNVZm

File: app.rb [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+
+  def create_teacher(age:, name:)
+    puts 'Please add your specialization:'
+    specialization = gets.chomp
+    teacher = Teacher.new(age: age, name: name, specialization: specialization)
+    @people << teacher
+    puts 'Teacher created successfully!!'
+  end
```
Comment (julie-ify) [inline app.rb:78]: - [ ] The app crashes with the attached error message when I tried creating a teacher. Kindly fix this to make your app work as expected.
chrome_4Sipp1CKoH

File: app.rb [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+      renter_id = gets.chomp.to_i
+      renter = @people.select { |person| person.id == renter_id }.first
+
+      print 'Enter the date[yyyy-mm-dd]: '
+      date = gets.chomp.to_s
+      rental = Rental.new(date: date, person: renter, book: choosen_book)
+      @rentals << rental
+      puts 'Book rented successfully!!'
+    end
+  end
```
Comment (julie-ify) [inline app.rb:119]: - [ ]  The app crashes with the attached error message when I tried creating a rental. Kindly fix this to make your app work as expected.

chrome_08aQLgdo5B

**Review by julie-ify — CHANGES_REQUESTED** [review]
## Hi @joel-kalema,

Good job so far!

## Highlights!
- Good feature branch name ✔️ 
- No linter errors ✔️   

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app.rb (hunk 1 above)
Comment (ahmetbozaci) [reply app.rb:70]: **Well done fix the error.** 🥇

File: app.rb (hunk 2 above)
Comment (ahmetbozaci) [reply app.rb:78]: **Well done fix the error.** 🥇

File: app.rb (hunk 3 above)
Comment (ahmetbozaci) [reply app.rb:119]: - [ ] I am getting an error when trying to create the rental kindly fix it.


image

**Review by ahmetbozaci — CHANGES_REQUESTED** [review]
# STATUS: Changes Required  ♻️


@joel-kalema  ✋🏼 
#### Great work on making the changes requested by a previous reviewer 👏 
#### You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

- Your git-flow is correct. ✔️
- You set up linter files and no errors. ✔️
- Your pull request is very professional. It has a meaningful title and summary. Keep it up.✔️
___
### ♻️ Request changes ♻️

- [ ] _Check the comments under the review._

___

### Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
___

### Cheers and Happy coding! 💻 👏
almost

___

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification._

_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._
___

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form.](https://airtable.com/shrQAqnBwek5a0O0s)_

[GitHub](https://github.com/ahmetbozaci) | [LinkedIn](https://linkedin.com/in/ahmetbozaci) | [Twitter](https://twitter.com/ahmtbozaci)

_Reviewed by [Ahmet Bozacı ](https://github.com/ahmetbozaci)_

**Comment by joel-kalema** [comment]
ohhh,...thank you for your feedback, I'll do my best to fix it 👍

**Review by rotimiazeez — APPROVED** [review]
# Hi @joel-kalema 👍🏽 , 

### Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**